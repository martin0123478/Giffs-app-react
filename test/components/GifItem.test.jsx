import { render, screen } from '@testing-library/react'
import GifItem from '../../src/components/GifItem'
describe('Pruebas en GifItem.js', () => {
    const title = 'saitama'
    const url = 'https://one-punch/saitama.jpg'
    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })

    test('Debe mostrar la imagen con el URL y el alt indicado', () => {
        render(<GifItem title={title} url={url} />)
        //screen.debug()
        //  expect(screen.getByRole('img').src).toBe(url)
        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)

    })

    test('debe mostar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />)
        expect(screen.getByAltText(title)).toBeTruthy()
    })
})