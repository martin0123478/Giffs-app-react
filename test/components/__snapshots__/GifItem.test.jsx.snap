// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem.js Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://one-punch/saitama.jpg"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
